server/pulp/server/managers/content/orphan.py

```diff
@@ -256,13 +259,42 @@ def delete_orphan_content_units_by_type(type_id):
             for non_orphan_id in non_orphan:
                 unit_dict.pop(non_orphan_id)
 
-            # Remove the unit and any references on disk
+            # If there's a list of unit ids to delete, filter the orphan list
```

Comment:
Why not do the filtering on line 245 above? It looks like that could avoid loading a lot of data.